indexer/feature_meta.cpp

```diff
@@ -1 +1,72 @@
 #include "indexer/feature_meta.hpp"
+
+namespace
+{
+char HexToDec(char ch)
+{
+  if (ch >= '0' && ch <= '9')
+    return ch - '0';
+  if (ch >= 'a')
+    ch -= 'a' - 'A';
+  if (ch >= 'A' && ch <= 'F')
+    return ch - 'A' + 10;
+  return -1;
+}
+
+string UriDecode(string const & sSrc)
+{
+  // This code is based on
+  // http://www.codeguru.com/cpp/cpp/string/conversions/article.php/c12759
+  //
+  // Note from RFC1630: "Sequences which start with a percent
+  // sign but are not followed by two hexadecimal characters
+  // (0-9, A-F) are reserved for future extension"
+
+  string result(sSrc.length(), 0);
+  auto itResult = result.begin();
+
+  for (auto it = sSrc.begin(); it != sSrc.end(); ++it)
+  {
+    if (*it == '%')
+    {
+      if (distance(it, sSrc.end()) > 2)
+      {
+        char dec1 = HexToDec(*(it + 1));
```

Comment:
The one there is a different one. For example, I need only to decode one 2-byte sequence, instead of a whole hexadecimal string. And wrong digits are acceptable here — these are just not decoded.